src/Utils.js

```diff
@@ -189,7 +189,7 @@ export function selectorError(selector, type = '') {
   );
 }
 
-export const isCompoundSelector = /([a-z]\.[a-z]|[a-z]\[.*\]|[a-z]#[a-z])/i;
+export const isCompoundSelector = /^[\.#]?-?[_a-z]+[_a-z0-9-]*[\.\[#]/i;
```

Comment:
@lencioni we have an open PR for implementing a parser https://github.com/airbnb/enzyme/pull/458